core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -189,7 +190,10 @@ protected ShardValidateQueryResponse shardOperation(ShardValidateQueryRequest re
             }
             if (request.rewrite()) {
                 explanation = getRewrittenQuery(searcher.searcher(), searchContext.query());
-            }   
+            }
+        } catch (QueryShardException e) {
+            valid = false;
+            error = e.getDetailedMessage();
         } catch (QueryParsingException e) {
```

Comment:
Why? As far as I can see queryParserService.parseQuery(request.source()) can still throw QueryParsingException, I didn't change that exception there because that method itself does some xContent parsing.